### ISSUE TITLE
Tests fail on pytest 6

### ISSUE THREAD

**Issue by eli-schwartz** [pytest-dev/pytest-randomly#305]
See https://docs.pytest.org/en/stable/deprecations.html#node-construction-changed-to-node-from-parent

While doing a mass rebuild for python 3.9, https://www.archlinux.org/packages/community/any/python-pytest-randomly/ failed to pass the testsuite with the following error log:

```
============================= test session starts ==============================
platform linux -- Python 3.9.0, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
Using --randomly-seed=610203316
rootdir: /build/python-pytest-randomly/src/pytest-randomly-3.4.1
plugins: randomly-3.4.1, Faker-4.14.2, forked-1.3.0, xdist-2.1.0
collected 34 items / 1 deselected / 33 selected

tests/test_pytest_randomly.py .....................F...........          [100%]

=================================== FAILURES ===================================
__________________ test_it_works_with_the_simplest_test_items __________________

ourtestdir = <Testdir local('/tmp/pytest-of-builduser/pytest-0/test_it_works_with_the_simplest_test_items0')>

    def test_it_works_with_the_simplest_test_items(ourtestdir):
        ourtestdir.makepyfile(
            conftest="""
            import pytest
    
    
            class MyCollector(pytest.Collector):
                def __init__(self, fspath, items, **kwargs):
                    super(MyCollector, self).__init__(fspath, **kwargs)
                    self.items = items
    
                def collect(self):
                    return self.items
    
    
            class NoOpItem(pytest.Item):
                def __init__(self, path, parent, module=None):
                    super(NoOpItem, self).__init__(path, parent)
                    if module is not None:
                        self.module = module
    
                def runtest(self):
                    pass
    
    
            def pytest_collect_file(path, parent):
                if not str(path).endswith('.py'):
                    return
                return MyCollector(
                    fspath=str(path),
                    items=[
                    NoOpItem(str(path), parent, 'foo'),
                    NoOpItem(str(path), parent),
                    ],
                    parent=parent,
                )
            """
        )
        args = ["-v"]
    
        out = ourtestdir.runpytest(*args)
>       out.assert_outcomes(passed=2)
E       AssertionError: assert {'errors': 1,...pped': 0, ...} == {'errors': 0,...pped': 0, ...}
E         Omitting 4 identical items, use -vv to show
E         Differing items:
E         {'errors': 1} != {'errors': 0}
E         {'passed': 0} != {'passed': 2}
E         Use -v to get the full diff

/build/python-pytest-randomly/src/pytest-randomly-3.4.1/tests/test_pytest_randomly.py:467: AssertionError
----------------------------- Captured stdout call -----------------------------
============================= test session starts ==============================
platform linux -- Python 3.9.0, pytest-6.1.2, py-1.9.0, pluggy-0.13.1 -- /usr/bin/python
cachedir: .pytest_cache
Using --randomly-seed=610203316
rootdir: /tmp/pytest-of-builduser/pytest-0/test_it_works_with_the_simplest_test_items0, configfile: pytest.ini
plugins: randomly-3.4.1, Faker-4.14.2, forked-1.3.0, xdist-2.1.0
collecting ... collected 0 items / 1 error

==================================== ERRORS ====================================
________________________ ERROR collecting test session _________________________
Direct construction of NoOpItem has been deprecated, please use NoOpItem.from_parent.
See https://docs.pytest.org/en/stable/deprecations.html#node-construction-changed-to-node-from-parent for more details.
=========================== short test summary info ============================
ERROR 
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.08s ===============================
=========================== short test summary info ============================
FAILED tests/test_pytest_randomly.py::test_it_works_with_the_simplest_test_items
================== 1 failed, 32 passed, 1 deselected in 9.67s ==================
```

**Comment by adamchainz** [comment]
Officially I only support running the tests with the setup in the repo, using `tox`. However problem was fixed in #281, and wasn't released yet. Just released in version 3.5.0.

**Comment by eli-schwartz** [comment]
Thanks for the release!

... and apologies for not noticing that a fix was in git master but unreleased. I thought I checked, but I must not have spotted it.
